#!/usr/bin/env python
PACKAGE = "fub_visual_odometry"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("car_front_marker_min_h", int_t,    0, "Min hue for the car marker", 72,  0, 255)
gen.add("car_front_marker_max_h", int_t,    0, "Max hue for the car marker", 146,  0, 255)
gen.add("car_front_marker_min_s", int_t,    0, "Min saturation for the car marker", 147,  0, 255)
gen.add("car_front_marker_max_s", int_t,    0, "Max saturation for the car marker", 255,  0, 255)
gen.add("car_front_marker_min_v", int_t,    0, "Min value for the car marker", 076,  0, 255)
gen.add("car_front_marker_max_v", int_t,    0, "Max value for the car marker", 213,  0, 255)

gen.add("car_rear_marker_min_h", int_t,    0, "Min hue for the car marker", 72,  0, 255)
gen.add("car_rear_marker_max_h", int_t,    0, "Max hue for the car marker", 146,  0, 255)
gen.add("car_rear_marker_min_s", int_t,    0, "Min saturation for the car marker", 147,  0, 255)
gen.add("car_rear_marker_max_s", int_t,    0, "Max saturation for the car marker", 255,  0, 255)
gen.add("car_rear_marker_min_v", int_t,    0, "Min value for the car marker", 076,  0, 255)
gen.add("car_rear_marker_max_v", int_t,    0, "Max value for the car marker", 213,  0, 255)

gen.add("map_marker_min_h", int_t,    0, "Min hue for the map marker", 12,  0, 255)
gen.add("map_marker_max_h", int_t,    0, "Max hue for the map marker", 70,  0, 255)
gen.add("map_marker_min_s", int_t,    0, "Min saturation for the map marker", 147,  0, 255)
gen.add("map_marker_max_s", int_t,    0, "Max saturation for the map marker", 255,  0, 255)
gen.add("map_marker_min_v", int_t,    0, "Min value for the map marker", 48,  0, 255)
gen.add("map_marker_max_v", int_t,    0, "Max value for the map marker", 213,  0, 255)


exit(gen.generate(PACKAGE, "fub_visual_odometry", "VisualOdometry"))